scripts/protect_branch_git_hook.py

```diff
@@ -0,0 +1,28 @@
+import subprocess
+import sys
+
+
+def ProtectMasterBranch():
```

Comment:
For the sake of enforcing good habits, this should have a return type annotations. In this case `NoReturn` as all branches of this function end with `exit()`, which is a `NoReturn` type of function as well.

Reply:
Also, let's rename to something more generic e.g. `ProtectBranches` since there are now multiple branches to protect (`main`, and all the `branch-x.y` as well)